### PR TITLE
Data cleaning (health and GtR)

### PR DESCRIPTION
**Pull request by jaklinger**
The main change is the introduction of `ElasticsearchPlus` which chains together a large number of rule-based cleaning procedures.

After converting the pipelines to use this, a number of enhancements/bug fixes from the original pipeline were implemented (character encoding in the docker image, automatic reindexing if specified for test tasks)

A description of updates can be found here: https://github.com/nestauk/nesta/issues/81

GtR update is to split out "organisations" and "participants", so that a new table has been introduced which provides more information to the user.

### CODE REVIEW

File: nesta/production/batchables/health_data/nih_process_data/run.py [hunk 1]
```diff
@@ -56,23 +59,38 @@ def run():
 
     # clean start and end dates
     for col in ["project_start", "project_end"]:
-        df[col] = df[col].apply(_extract_date)
+        df[col] = df[col].apply(lambda x: _extract_date(x)) #, return_date_object=True))
```
Comment (russwinch) [inline nesta/production/batchables/health_data/nih_process_data/run.py]: Can this commented out bit be removed?

File: nesta/production/orms/orm_utils.py [hunk 2]
```diff
@@ -6,35 +6,144 @@
...
+
+
+def setup_es(es_mode, test_mode, reindex_mode, dataset, aliases=None):
+    """Retrieve the ES connection, ES config and setup the index 
+    if required.
+    
+    Args:
+        es_mode (str): One of "prod" or "dev".
+        test_mode (bool): Running in test mode?
+        reindex_mode (bool): Running in reindex mode?
```
Comment (russwinch) [inline nesta/production/orms/orm_utils.py]: I know this is minor, but could this be called something else so it's not confused with ES reindexing process? `recreate_index` / `drop_and_recreate`?

File: nesta/production/batchables/health_data/nih_process_data/run.py (hunk 1 above)
Comment (jaklinger) [reply nesta/production/batchables/health_data/nih_process_data/run.py]: done (see in new commit)

File: nesta/production/orms/orm_utils.py (hunk 2 above)
Comment (jaklinger) [reply nesta/production/orms/orm_utils.py]: good idea - changed in favour of `drop_and_recreate`

**Pull request merged**